server/container_create_linux.go

```diff
@@ -411,6 +414,28 @@ func (s *Server) mountImage(ctx context.Context, specgen *generate.Generator, im
 
 	log.Infof(ctx, "Image mounted to: %s", mountPoint)
 
+	if m.ImageSubPath != "" {
+		// Kubernetes already verifies that ImageSubPath:
+		// - is a relative path
+		// - does not contain any '..'
+		//
+		// We cannot create non-existing paths within the image volume because
+		// they're mounted read-only.
+		mountPointFile, err := securejoin.OpenInRoot(mountPoint, m.ImageSubPath)
+		if err != nil {
+			// This is a user-facing error message from a kubelet event, means we should make it as meaningful as possible.
+			if errors.Is(err, unix.ENOENT) {
+				return nil, fmt.Errorf("sub path %q does not exist in image volume %q", m.ImageSubPath, m.Image.Image)
+			}
+
+			return nil, fmt.Errorf("join final mount path: %w", err)
+		}
+		defer mountPointFile.Close()
+
+		mountPoint = mountPointFile.Name()
```

Comment:
this returns the full path to the original file, right?  It is not safe to use, we need the `/proc/self/fd/$FD` path

Reply:
and the problem is that the fd must be kept open until it is referenced

Reply:
@giuseppe can we make the `safeMountSubPath` a helper in c/common or similar?